### PR TITLE
DS-5264: Send errors to channel only when non-nil. Improve logging.

### PR DESCRIPTION
**Pull request by jindrazak**
**What type of PR is this?**
bug

**What this PR does / why we need it**:
Sometimes, we observed a nil pointer dereference error when executing a command with `cmdexecutor` in multiple pods:

```
INFO[0000] Running pod command executor:
INFO[0000] Using timeout: 86400 seconds
INFO[0004] Running command: pds-backup.sh --type=snapshot && ((sleep 10; touch /tmp/killme-5748ec60-44d2-40b4-a74b-3abbfe18ea8a)&) && /tmp/wait-5748ec60-44d2-40b4-a74b-3abbfe18ea8a.sh on pod: [dev] cas-jzak-krv5xr-0
INFO[0005] Running command: pds-backup.sh --type=snapshot && ((sleep 10; touch /tmp/killme-5748ec60-44d2-40b4-a74b-3abbfe18ea8a)&) && /tmp/wait-5748ec60-44d2-40b4-a74b-3abbfe18ea8a.sh on pod: [dev] cas-jzak-krv5xr-1
INFO[0007] Checking status on command: pds-backup.sh --type=snapshot && ((sleep 10; touch /tmp/killme-5748ec60-44d2-40b4-a74b-3abbfe18ea8a)&) && ${WAIT_CMD}
INFO[0007] Running command: pds-backup.sh --type=snapshot && ((sleep 10; touch /tmp/killme-5748ec60-44d2-40b4-a74b-3abbfe18ea8a)&) && /tmp/wait-5748ec60-44d2-40b4-a74b-3abbfe18ea8a.sh on pod: [dev] cas-jzak-krv5xr-2
INFO[0007] check status on pod: [dev] cas-jzak-krv5xr-2 with backoff: {2s 1 0.1 43200 0s} and status file: /tmp/stork-cmd-done-5748ec60-44d2-40b4-a74b-3abbfe18ea8a
INFO[0007] check status on pod: [dev] cas-jzak-krv5xr-1 with backoff: {2s 1 0.1 43200 0s} and status file: /tmp/stork-cmd-done-5748ec60-44d2-40b4-a74b-3abbfe18ea8a
INFO[0007] check status on pod: [dev] cas-jzak-krv5xr-0 with backoff: {2s 1 0.1 43200 0s} and status file: /tmp/stork-cmd-done-5748ec60-44d2-40b4-a74b-3abbfe18ea8a
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x13f316c]

goroutine 1 [running]:
main.main()
        /home/jzak/projects/stork/cmd/cmdexecutor/cmdexecutor.go:181 +0x8ac

Process finished with the exit code 2
```

This error is triggered by calling `err.Error()` when the `err` is nil at https://github.com/libopenstorage/stork/blob/master/cmd/cmdexecutor/cmdexecutor.go#L181. This is caused by sending nil pointer to the error channel.

I also added a couple of log outputs so that it's clearer what's the current state. Log output with these changes:

```INFO[0001] Running pod command executor:
INFO[0001] Using timeout: 86400 seconds
INFO[0005] Running command: pds-backup.sh --type=snapshot && /tmp/wait-5748ec60-44d2-40b4-a74b-3abbfe18ea8a.sh on pod: [dev] cas-jzak-krv5xr-0
INFO[0007] Running command: pds-backup.sh --type=snapshot && /tmp/wait-5748ec60-44d2-40b4-a74b-3abbfe18ea8a.sh on pod: [dev] cas-jzak-krv5xr-1
INFO[0009] Checking status on command: pds-backup.sh --type=snapshot && ${WAIT_CMD}
INFO[0009] Running command: pds-backup.sh --type=snapshot && /tmp/wait-5748ec60-44d2-40b4-a74b-3abbfe18ea8a.sh on pod: [dev] cas-jzak-krv5xr-2
INFO[0009] check status on pod: [dev] cas-jzak-krv5xr-1 with backoff: {2s 1 0.1 43200 0s} and status file: /tmp/stork-cmd-done-5748ec60-44d2-40b4-a74b-3abbfe18ea8a
INFO[0009] check status on pod: [dev] cas-jzak-krv5xr-2 with backoff: {2s 1 0.1 43200 0s} and status file: /tmp/stork-cmd-done-5748ec60-44d2-40b4-a74b-3abbfe18ea8a
INFO[0009] check status on pod: [dev] cas-jzak-krv5xr-0 with backoff: {2s 1 0.1 43200 0s} and status file: /tmp/stork-cmd-done-5748ec60-44d2-40b4-a74b-3abbfe18ea8a
INFO[0011] checked status on pod: [dev] cas-jzak-krv5xr-1 with result: job not finished yet, will retry later
INFO[0011] checked status on pod: [dev] cas-jzak-krv5xr-0 with result: job not finished yet, will retry later
INFO[0011] checked status on pod: [dev] cas-jzak-krv5xr-2 with result: job not finished yet, will retry later
INFO[0015] checked status on pod: [dev] cas-jzak-krv5xr-1 with result: job not finished yet, will retry later
INFO[0015] checked status on pod: [dev] cas-jzak-krv5xr-2 with result: job not finished yet, will retry later
INFO[0015] checked status on pod: [dev] cas-jzak-krv5xr-0 with result: job not finished yet, will retry later
INFO[0019] checked status on pod: [dev] cas-jzak-krv5xr-1 with result: job not finished yet, will retry later
INFO[0019] checked status on pod: [dev] cas-jzak-krv5xr-2 with result: job not finished yet, will retry later
INFO[0019] checked status on pod: [dev] cas-jzak-krv5xr-0 with result: job not finished yet, will retry later
INFO[0022] checked status on pod: [dev] cas-jzak-krv5xr-2 with result: job not finished yet, will retry later
INFO[0022] checked status on pod: [dev] cas-jzak-krv5xr-1 with result: job not finished yet, will retry later
INFO[0023] checked status on pod: [dev] cas-jzak-krv5xr-0 with result: job not finished yet, will retry later
INFO[0026] checked status on pod: [dev] cas-jzak-krv5xr-2 with result: job not finished yet, will retry later
INFO[0027] checked status on pod: [dev] cas-jzak-krv5xr-1 with result: job not finished yet, will retry later
INFO[0027] checked status on pod: [dev] cas-jzak-krv5xr-0 with result: job finished
INFO[0029] finished waiting for pod, 2 pods left to finish
INFO[0030] checked status on pod: [dev] cas-jzak-krv5xr-1 with result: job not finished yet, will retry later
INFO[0031] checked status on pod: [dev] cas-jzak-krv5xr-2 with result: job not finished yet, will retry later
INFO[0034] checked status on pod: [dev] cas-jzak-krv5xr-2 with result: job not finished yet, will retry later
INFO[0034] checked status on pod: [dev] cas-jzak-krv5xr-1 with result: job finished
INFO[0036] finished waiting for pod, 1 pods left to finish
INFO[0038] checked status on pod: [dev] cas-jzak-krv5xr-2 with result: job finished
INFO[0041] finished waiting for pod, 0 pods left to finish
INFO[0041] successfully executed command: pds-backup.sh --type=snapshot && ${WAIT_CMD} on all pods: [dev/cas-jzak-krv5xr-0 dev/cas-jzak-krv5xr-1 dev/cas-jzak-krv5xr-2]
```

**Does this PR change a user-facing CRD or CLI?**:

No

**Is a release note needed?**:
I guess not.

**Does this change need to be cherry-picked to a release branch?**:
No

After merging, let's upgrade to the newer version in PDS as part of DS-5354.

### CODE REVIEW

File: pkg/cmdexecutor/cmdexecutor.go [hunk 1]
```diff
@@ -90,9 +90,8 @@ func (c *cmdExecutor) Start(errChan chan error) error {
 			err = fmt.Errorf("failed to run command: %s in pod: [%s] %s due to err: %v",
 				command, c.podNamespace, c.podName, err)
 			logrus.Errorf(err.Error())
+			errChan <- err
 		}
-
-		errChan <- err
```
Comment (jindrazak) [inline pkg/cmdexecutor/cmdexecutor.go:95]: Let's send the `err` to the channel only if it's not `nil`. This is what fixes the nil pointer dereference error.

**Comment by jindrazak** [comment]
Please, let me know if there's any other information you need from us regarding the PR.

**Comment by adityadani** [comment]
@siva-portworx Can you also take a look at this PR ?

**Review by siva-portworx — APPROVED** [review]
LGTM

**Comment by siva-portworx** [comment]
Only the codecov is failing So force merging it.

**Pull request merged**